Gemfile

```diff
@@ -2,86 +2,37 @@
 
 source 'https://rubygems.org'
 
-group :backend, :frontend, :core, :api do
-  gem 'solidus_api', github: 'solidusio/solidus', glob: '**/*.gemspec'
-  gem 'solidus_backend', github: 'solidusio/solidus', glob: '**/*.gemspec'
-  gem 'solidus_core', github: 'solidusio/solidus', glob: '**/*.gemspec'
-  gem 'solidus_frontend', github: 'solidusio/solidus', glob: '**/*.gemspec'
-  gem 'solidus_sample', github: 'solidusio/solidus', glob: '**/*.gemspec'
+branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
 
-  # rubocop:disable Bundler/DuplicatedGem
-  if ENV['RAILS_VERSION'] == 'master'
-    gem 'rails', github: 'rails', require: false
-  else
-    gem 'rails', ENV['RAILS_VERSION'] || '~> 7.0.2', require: false
-  end
-  # rubocop:enable Bundler/DuplicatedGem
-
-  # Temporarily locking sprockets to v3.x
-  # see https://github.com/solidusio/solidus/issues/3374
-  # and https://github.com/rails/sprockets-rails/issues/369
-  gem 'sprockets', '~> 3'
-
-  platforms :ruby do
-    if /mysql/.match?(ENV['DB']) || ENV['DB_ALL']
-      gem 'mysql2', '~> 0.5.0', require: false
-    end
-    if /postgres/.match?(ENV['DB']) || ENV['DB_ALL']
-      gem 'pg', '~> 1.0', require: false
-    end
-    if ENV['DB_ALL'] || !/mysql|postgres/.match?(ENV['DB'])
-      gem 'sqlite3', require: false
-      gem 'fast_sqlite', require: false
-    end
-  end
-
-  platforms :jruby do
-    gem 'jruby-openssl', require: false
-    gem 'activerecord-jdbcsqlite3-adapter', require: false
-  end
-
-  gem 'database_cleaner', '~> 1.3', require: false
-  gem 'rspec-activemodel-mocks', '~> 1.1', require: false
-  gem 'rspec-rails', '~> 4.0.1', require: false
-  gem 'simplecov', require: false
-  gem 'with_model', require: false
-  gem 'rails-controller-testing', require: false
-  gem 'puma', require: false
-
-  # Ensure the requirement is also updated in core/lib/spree/testing_support/factory_bot.rb
-  gem 'factory_bot_rails', '>= 4.8', require: false
-end
-
-group :backend, :frontend do
-  gem 'capybara', '~> 3.13', require: false
-  gem 'capybara-screenshot', '>= 1.0.18', require: false
-  gem 'selenium-webdriver', require: false
-  gem 'webdrivers', require: false
-end
-
-group :frontend do
-  gem 'generator_spec'
+git "https://github.com/solidusio/solidus.git", branch: branch do
+  gem 'solidus_api'
+  gem 'solidus_core'
```

Comment:
I originally wrote the Solidus gems like this:

```
  gem 'solidus_api', github: 'solidusio/solidus', glob: 'api/solidus_api.gemspec'
  gem 'solidus_core', github: 'solidusio/solidus', glob: 'core/solidus_core.gemspec'
```

However, I encountered this issue in CI, where the Bundler version used is older (1.17): https://github.com/rubygems/bundler/issues/7346.

Wrapping the gems in a git block works around the issue.